### ISSUE TITLE
`payments/accounts` returns pending_requirements state after deadline

### ISSUE THREAD

**Issue by malinajirka** [Automattic/woocommerce-payments#2676]
`payments/accounts` endpoint returns `has_overdue_requirements=false` even when the `current_deadline` date is in the past. If I understand the fields correctly it should return `restricted` instead of `restricted_soon` and return `has_overdue_requirements=true`, right?

```
data: {
 card_present_eligible: true
 country: "US"
 current_deadline: 1627653583
 has_overdue_requirements: false
 has_pending_requirements: true
 statement_descriptor: "WOOWCPAYSTRIPEACCOUNTP"
 status: "restricted_soon"
▼store_currencies: {
 default: "usd"
▶supported: ["usd"]
}
}
```

I tested this on site with .com id `196011178`. 

cc @allendav @kalessil @koke @ctarda

**Comment by kalessil** [comment]
`has_overdue_requirements` is populated from account data, fetched from Stripe API. This property used: https://stripe.com/docs/api/accounts/object#account_object-requirements-past_due. So that is something we can ask Stripe (will do tomorrow).

`restricted_soon`: calculated based on other account data and we can check details, is that the account for the Store you shared early today?

**Comment by malinajirka** [comment]
> is that the account for the Store you shared early today?

The credentials I shared first time but they did not work are the correct one -> they are for a .com account. The site I'm able to reproduce it on is on that account - it's called woowcpaystripeaccountpendingrequirementstestingsite.

> So that is something we can ask Stripe (will do tomorrow).

Perfect, thanks!

**Comment by kalessil** [comment]
Asked here: p1628234395051100-slack-C9976E5MJ

**Comment by kalessil** [comment]
Side-note: Stripe dashboard shows the account status `Restricted soon` as well.

**Comment by htdat** [comment]
> Asked here: p1628234395051100-slack-C9976E5MJ

Can you summarize this convo and update the issue description? 

FYI. I am doing triage for WCPay repos and ensure that the issue is clear and ready for work.

**Comment by htdat** [comment]
> Asked here: p1628234395051100-slack-C9976E5MJ

Can you summarize this convo and update the issue description? 

FYI. I am doing triage for WCPay repos and ensure that the issue is clear and ready for work.

**Comment by jessy-p** [comment]
According to the conversation linked above, this is the expected behavior. 
`has_overdue_requirements` from Stripe's [past_due](https://stripe.com/docs/api/accounts/object#account_object-requirements-past_due) is empty because the due date and the grace period of 7 days to update the information has passed.
